### ISSUE TITLE
Integer underflow in quinn-proto

### ISSUE THREAD

**Issue by jean-airoldie** [djc/quinn#846]
I'm hitting a underflow when using quinn.

<details><summary>here is the backtrace</summary>
<p>

```
thread '<unnamed>' panicked at 'attempt to subtract with overflow', /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/range_set.rs:158:42
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:483
   1: core::panicking::panic_fmt
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/panicking.rs:85
   2: core::panicking::panic
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/panicking.rs:50
   3: quinn_proto::range_set::RangeSet::max::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/range_set.rs:158
   4: core::option::Option<T>::map
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/option.rs:437
   5: quinn_proto::range_set::RangeSet::max
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/range_set.rs:158
   6: quinn_proto::streams::Recv::limit
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/streams.rs:453
   7: quinn_proto::connection::Connection<S>::process_payload
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/connection.rs:1803
   8: quinn_proto::connection::Connection<S>::process_decrypted_packet
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/connection.rs:1557
   9: quinn_proto::connection::Connection<S>::handle_packet
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/connection.rs:1297
  10: quinn_proto::connection::Connection<S>::handle_decode
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/connection.rs:1215
  11: quinn_proto::connection::Connection<S>::handle_event
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-proto-0.6.1/src/connection.rs:1148
  12: quinn::connection::ConnectionInner<S>::process_conn_events
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-0.6.1/src/connection.rs:673
  13: <quinn::connection::ConnectionDriver<S> as core::future::future::Future>::poll
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-0.6.1/src/connection.rs:233
  14: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/core.rs:173
  15: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/loom/std/unsafe_cell.rs:14
  16: tokio::runtime::task::core::Core<T,S>::poll
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/core.rs:158
  17: tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/harness.rs:107
  18: core::ops::function::FnOnce::call_once
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  19: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:308
  20: std::panicking::try::do_call
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:381
  21: __rust_try
  22: std::panicking::try
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:345
  23: std::panic::catch_unwind
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:382
  24: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/harness.rs:89
  25: tokio::runtime::task::raw::poll
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/raw.rs:104
  26: tokio::runtime::task::raw::RawTask::poll
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/raw.rs:66
  27: tokio::runtime::task::Notified<S>::run
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/mod.rs:169
  28: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:155
  29: tokio::coop::with_budget::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:127
  30: std::thread::local::LocalKey<T>::try_with
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/local.rs:265
  31: std::thread::local::LocalKey<T>::with
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/local.rs:241
  32: tokio::coop::with_budget
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:120
  33: tokio::coop::budget
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:96
  34: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:155
  35: tokio::runtime::basic_scheduler::enter::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:213
  36: tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/macros/scoped_tls.rs:63
  37: tokio::runtime::basic_scheduler::enter
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:213
  38: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:123
  39: tokio::runtime::Runtime::block_on::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/mod.rs:444
  40: tokio::runtime::context::enter
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/context.rs:72
  41: tokio::runtime::handle::Handle::enter
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/handle.rs:76
  42: tokio::runtime::Runtime::block_on
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/mod.rs:441
  43: smol::run::{{closure}}::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/smol-0.3.3/src/lib.rs:191
  44: scoped_tls::ScopedKey<T>::set
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
  45: async_executor::Executor::enter
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-0.1.2/src/lib.rs:151
  46: smol::run::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/smol-0.3.3/src/lib.rs:191
  47: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  48: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
  49: easy_parallel::Parallel<T>::finish::{{closure}}
             at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/easy-parallel-3.1.0/src/lib.rs:213
  50: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  51: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-0.6.1/src/connection.rs:595:40
stack backtrace:
   0:     0x559b109f1b20 - std::backtrace_rs::backtrace::libunwind::trace::hdf911925cfd1a062
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x559b109f1b20 - std::backtrace_rs::backtrace::trace_unsynchronized::h1a4f248df23e18fe
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x559b109f1b20 - std::sys_common::backtrace::_print_fmt::hf3ec23fc59b676bd
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:79
   3:     0x559b109f1b20 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h00fdb223b1b833ac
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:58
   4:     0x559b10a154ac - core::fmt::write::h1857a60b204f1b6a
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/fmt/mod.rs:1082
   5:     0x559b109eb487 - std::io::Write::write_fmt::ha851958ed267cb97
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/io/mod.rs:1514
   6:     0x559b109f4010 - std::sys_common::backtrace::_print::hf25dbce5d8adc35c
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:61
   7:     0x559b109f4010 - std::sys_common::backtrace::print::h40243408d050ccf7
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:48
   8:     0x559b109f4010 - std::panicking::default_hook::{{closure}}::ha452abcc85cd3549
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:208
   9:     0x559b109f3d5c - std::panicking::default_hook::h637245b92cbb00b1
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:227
  10:     0x559b109f4673 - std::panicking::rust_panic_with_hook::h2f4c96dfd8ba524a
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:577
  11:     0x559b109f4249 - std::panicking::begin_panic_handler::{{closure}}::h7740abbe2875cb4d
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:484
  12:     0x559b109f1fac - std::sys_common::backtrace::__rust_end_short_backtrace::hcad001df0a36db28
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:153
  13:     0x559b109f4209 - rust_begin_unwind
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:483
  14:     0x559b10a13d61 - core::panicking::panic_fmt::hb15d6f55e8472f62
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/panicking.rs:85
  15:     0x559b10a13b83 - core::option::expect_none_failed::he492a18657d97593
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/option.rs:1221
  16:     0x559b1047fc8f - core::result::Result<T,E>::unwrap::h94285c1262ddc86d
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/result.rs:973
  17:     0x559b103384b4 - <quinn::connection::ConnectionRef<S> as core::ops::drop::Drop>::drop::h16b59c308f75f287
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-0.6.1/src/connection.rs:595
  18:     0x559b103283d2 - core::ptr::drop_in_place::h66f67d8f9817ac4a
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ptr/mod.rs:184
  19:     0x559b1033077e - core::ptr::drop_in_place::heca1f9c776cdb688
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ptr/mod.rs:184
  20:     0x559b1032d902 - core::ptr::drop_in_place::hbea76a41bd4bc464
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ptr/mod.rs:184
  21:     0x559b103606fb - tokio::runtime::task::core::Core<T,S>::drop_future_or_output::{{closure}}::hee2c9b66273a07f7
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/core.rs:192
  22:     0x559b103af276 - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::h31c855afb55c63bf
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/loom/std/unsafe_cell.rs:14
  23:     0x559b10360565 - tokio::runtime::task::core::Core<T,S>::drop_future_or_output::h96c5bea77198c290
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/core.rs:190
  24:     0x559b1031f91d - <tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}::Guard<T,S> as core::ops::drop::Drop>::drop::hcda26c0107ae12f0
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/harness.rs:96
  25:     0x559b1032d3ae - core::ptr::drop_in_place::hbb15922a459e6761
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ptr/mod.rs:184
  26:     0x559b1044fac8 - tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}::h064e26db21af9fb9
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/harness.rs:114
  27:     0x559b10321270 - core::ops::function::FnOnce::call_once::he5cf5282b5e78196
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  28:     0x559b104997da - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hba77119c361c3603
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:308
  29:     0x559b1049d2bd - std::panicking::try::do_call::h78f024858fe05dbf
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:381
  30:     0x559b1049e13d - __rust_try
  31:     0x559b1049c625 - std::panicking::try::h440b998ed6082f21
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:345
  32:     0x559b1049986a - std::panic::catch_unwind::h0610ab62728e4409
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:382
  33:     0x559b1044e5b8 - tokio::runtime::task::harness::Harness<T,S>::poll::h1e96836135c61fac
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/harness.rs:89
  34:     0x559b102fca50 - tokio::runtime::task::raw::poll::hfdbc57a79d2a80c6
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/raw.rs:104
  35:     0x559b1060fb4f - tokio::runtime::task::raw::RawTask::poll::hd3d1de8318cddd16
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/raw.rs:66
  36:     0x559b10499221 - tokio::runtime::task::Notified<S>::run::hb3ca65899af4a4f3
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/task/mod.rs:169
  37:     0x559b1031708a - tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}::{{closure}}::hbc342ad715cf29ad
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:155
  38:     0x559b103b8648 - tokio::coop::with_budget::{{closure}}::h3c7e5afcbdd40448
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:127
  39:     0x559b10446be6 - std::thread::local::LocalKey<T>::try_with::h01175c81f0d5f84a
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/local.rs:265
  40:     0x559b10446274 - std::thread::local::LocalKey<T>::with::hc622b43d8c542b24
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/local.rs:241
  41:     0x559b10316d54 - tokio::coop::with_budget::hb2b0ca5237ea019f
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:120
  42:     0x559b10316d54 - tokio::coop::budget::h008a0e34fe02a406
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/coop.rs:96
  43:     0x559b10316d54 - tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}::h5992e618207bd739
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:155
  44:     0x559b1031748d - tokio::runtime::basic_scheduler::enter::{{closure}}::h4cd0bb42e7696ff0
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:213
  45:     0x559b104dbeaf - tokio::macros::scoped_tls::ScopedKey<T>::set::h99ea36b85a1b3f15
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/macros/scoped_tls.rs:63
  46:     0x559b10317366 - tokio::runtime::basic_scheduler::enter::hc451cca2d1e0a680
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:213
  47:     0x559b10316687 - tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::h5e189a87a759167f
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/basic_scheduler.rs:123
  48:     0x559b104d98e7 - tokio::runtime::Runtime::block_on::{{closure}}::h616603d1b1047fea
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/mod.rs:444
  49:     0x559b10483f10 - tokio::runtime::context::enter::h27eeb164af51eeaf
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/context.rs:72
  50:     0x559b1035543a - tokio::runtime::handle::Handle::enter::h23e73b985dcb7f32
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/handle.rs:76
  51:     0x559b104d9787 - tokio::runtime::Runtime::block_on::h644af8bca95dcc1a
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/runtime/mod.rs:441
  52:     0x559b104c9100 - smol::run::{{closure}}::{{closure}}::h6c93ae48ee58f709
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/smol-0.3.3/src/lib.rs:191
  53:     0x559b1030e664 - scoped_tls::ScopedKey<T>::set::h63c1e01d48dfca0e
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/scoped-tls-1.0.0/src/lib.rs:137
  54:     0x559b104015fb - async_executor::Executor::enter::h38a40abfcea68c9f
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-0.1.2/src/lib.rs:151
  55:     0x559b104c767b - smol::run::{{closure}}::h4dc9071b93d2b61b
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/smol-0.3.3/src/lib.rs:191
  56:     0x559b103203ae - core::ops::function::FnOnce::call_once{{vtable.shim}}::hcd5d2f973334839b
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  57:     0x559b103b81e8 - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hf580bc9c01d960d6
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
  58:     0x559b104c06c1 - easy_parallel::Parallel<T>::finish::{{closure}}::h17f085f4b9b86b38
                               at /home/h4xor/.cargo/registry/src/github.com-1ecc6299db9ec823/easy-parallel-3.1.0/src/lib.rs:213
  59:     0x559b103204de - core::ops::function::FnOnce::call_once{{vtable.shim}}::heb409bc44438c24d
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  60:     0x559b103b8178 - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h0cf3c646d2e21a2d
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
  61:     0x559b10381b03 - std::sys_common::backtrace::__rust_begin_short_backtrace::h98b59338bff9aea3
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys_common/backtrace.rs:137
  62:     0x559b104b6193 - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::h284200b5060633ed
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/mod.rs:458
  63:     0x559b104996c3 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h65d3a3ca7916522f
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:308
  64:     0x559b1049d729 - std::panicking::try::do_call::he1c2704c38475dae
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:381
  65:     0x559b1049e13d - __rust_try
  66:     0x559b1049c780 - std::panicking::try::h6bb36b8865703d55
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panicking.rs:345
  67:     0x559b10499b93 - std::panic::catch_unwind::h96e761fe8db59d6f
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/panic.rs:382
  68:     0x559b104b5f9a - std::thread::Builder::spawn_unchecked::{{closure}}::h2c47627e3e394f7c
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/thread/mod.rs:457
  69:     0x559b103202be - core::ops::function::FnOnce::call_once{{vtable.shim}}::h92914b988b4ca5dd
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/core/src/ops/function.rs:227
  70:     0x559b109fb28a - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hd50938a7cbd66861
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
  71:     0x559b109fb28a - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hbdaa88962d74e8ba
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/alloc/src/boxed.rs:1042
  72:     0x559b109fb28a - std::sys::unix::thread::Thread::new::thread_start::h2e71bf448b68bc62
                               at /rustc/d006f5734f49625c34d6fc33bf6b9967243abca8/library/std/src/sys/unix/thread.rs:87
  73:     0x7f23b5396669 - start_thread
  74:     0x7f23b52a42b3 - __clone
  75:                0x0 - <unknown>
thread panicked while panicking. aborting.
```

</p>
</details>

I'll try to add reproduction code.

**Comment by jean-airoldie** [comment]
Fixed on master (4e819af57e179d32e8c13a8ac9210ec5093d20d2).

**Comment by Ralith** [comment]
That was probably addressed in 9cf8652deb89f16e236bc3d4fe46a42793a4c036 specifically. Thanks for the report, though! Would a patch release with the fix be helpful?

**Comment by jean-airoldie** [comment]
I don't mind using the master branch until the next release.
